controller/cmd/service-mirror/cluster_watcher.go

```diff
@@ -497,6 +500,132 @@ func remapRemoteServicePorts(ports []corev1.ServicePort) []corev1.ServicePort {
 	return newPorts
 }
 
+func (rcsw *RemoteClusterServiceWatcher) updateGatewayProxyService(spec *GatewaySpec) error {
```

Comment:
calling these gateway proxies is pretty confusing to me.  can we call them "gateway mirrors" or something?

Reply:
Better yet, add a comment that clarifies what it actually is?